lib/index.js

```diff
@@ -213,36 +218,61 @@ Mongoose.prototype.setDriver = function setDriver(driver) {
  * - 'toJSON': `{ transform: true, flattenDecimals: true }` by default. Overwrites default objects to [`toJSON()`](/docs/api.html#document_Document-toJSON), for determining how Mongoose documents get serialized by `JSON.stringify()`
  * - 'toObject': `{ transform: true, flattenDecimals: true }` by default. Overwrites default objects to [`toObject()`](/docs/api.html#document_Document-toObject)
  *
- * @param {String} key
- * @param {String|Function|Boolean} value
+ * @param {String|Object} key The name of the option or a object of multiple key-value pairs
+ * @param {String|Function|Boolean} value The value of the option, unused if "key" is a object
+ * @returns {Mongoose} The used Mongoose instnace
  * @api public
  */
 
 Mongoose.prototype.set = function(key, value) {
   const _mongoose = this instanceof Mongoose ? this : mongoose;
 
-  if (VALID_OPTIONS.indexOf(key) === -1) {
-    throw new Error(`\`${key}\` is an invalid option.`);
+  if (arguments.length === 1 && typeof key !== 'object') {
+    return _mongoose.options[key];
   }
 
-  if (arguments.length === 1) {
-    return _mongoose.options[key];
+  let options = {};
+
+  if (arguments.length === 2) {
+    options = { [key]: value };
   }
 
-  _mongoose.options[key] = value;
+  if (arguments.length === 1 && typeof key === 'object') {
+    options = key;
+  }
 
-  if (key === 'objectIdGetter') {
-    if (value) {
-      Object.defineProperty(mongoose.Types.ObjectId.prototype, '_id', {
-        enumerable: false,
-        configurable: true,
-        get: function() {
-          return this;
-        }
-      });
-    } else {
-      delete mongoose.Types.ObjectId.prototype._id;
+  // array for errors to collect all errors for all key-value pairs, like ".validate"
+  const errors = [];
+
+  for (const [k, v] of Object.entries(options)) {
+    if (VALID_OPTIONS.indexOf(k) === -1) {
+      errors.push(new Error(`\`${k}\` is an invalid option.`));
+      continue;
+    }
+
+    _mongoose.options[k] = v;
+
+    if (k === 'objectIdGetter') {
+      if (v) {
+        Object.defineProperty(mongoose.Types.ObjectId.prototype, '_id', {
+          enumerable: false,
+          configurable: true,
+          get: function() {
+            return this;
+          }
+        });
+      } else {
+        delete mongoose.Types.ObjectId.prototype._id;
+      }
+    }
+  }
+
+  if (errors.length) {
+    // this is to keep backwards-compat
+    if (errors.length === 1) {
+      throw errors[0];
     }
+    throw errors;
```

Comment:
[We should not throw an array](https://eslint.org/docs/latest/rules/no-throw-literal), because that does not include the stack trace, for example, even though each individual error has it.

For that, we'd have to create a new error that aggregates all the errors and adds them as to a single property. Look at `lib/error/validation.js` to see how we aggregate multiple validation errors into a single error object with a concatenated message and an `errors` property, including the individual errors.

Reply:
if you mention this rule and want to actually enforce it, maybe consider adding it to the current linting rules, because it seems like it is not there yet or by default

Reply:
I don't know if ESLint will enforce it against arrays or not, but feel free to add it to this PR as well if it's not part of the `eslint/recommended` rules.

Reply:
actually, i just looked, it is already in the eslint config, but it seems like it did not catch that (it is set to `error`)

Reply:
Yep, I don't think that ESLint is smart enough to pick up the types from variables.
It protects against `throw "Something bad happened.";` for example, but not other more complicated cases.

Reply:
fixed with 4ab2e442c4eeb50c1cd11407a10a5017da16ba05